server/eclipse-project/src/main/java/us/freeandfair/corla/asm/AbstractStateMachine.java

```diff
@@ -217,6 +228,16 @@ public ASMState stepTransition(final ASMTransition the_transition)
    */
   public ASMState stepEvent(final ASMEvent the_event)
       throws IllegalStateException {
+    // refresh and skip events are always permitted
```

Comment:
I don't like it either, but we gotta get this done.  I'll file a ticket for a refactor.